diff --git a/google3/third_party/jsoncpp/json_value.cc b/google3/third_party/jsoncpp/json_value.cc
index 4ba0200..868d6cc 100644
--- a/google3/third_party/jsoncpp/json_value.cc
+++ b/google3/third_party/jsoncpp/json_value.cc
@@ -39,8 +39,10 @@ namespace Json {
 // static
 Value const& Value::nullSingleton()
 {
- static Value const nullStatic;
- return nullStatic;
+  // Google3 specific: intentionally leak nullStatic to avoid TSan hits when
+  // global destructors are invoked with threads still running.
+  static const Value* const nullStatic = new Value;
+  return *nullStatic;
 }
 
 // for backwards compatibility, we'll leave these global references around, but DO NOT
